corehq/apps/app_manager/tasks.py

```diff
@@ -41,3 +43,21 @@ def create_build_files_for_all_app_profiles(domain, build_id):
             save_app = True
     if save_app:
         app.save()
+
+
+@task(queue='background_queue')
+def prune_auto_generated_builds(domain, app_id):
+    last_build = get_latest_build_id(domain, app_id)
+    query = (AppES()
+             .domain(domain)
+             .is_build()
+             .is_released(False)
+             .term('is_auto_generated', True)
+             .term('copy_of', app_id)
+             .source(['_id']))
+
+    for hit in query.run().hits:
+        if hit['_id'] == last_build:
+            continue
+        app = get_app(domain, hit['_id'])
+        app.delete()
```

Comment:
i usually like to double-check all the important things right before calling .delete(), just to make sure that if anything ever went wrong with the query (like if it returned the wrong results), that it wouldn't delete the wrong things. something like:

    if not app.is_auto_generated and ...:
        raise ValueError("...")

on a related note, is it possible to add a test for this?